Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -4725,6 +4726,85 @@ RefPtr<CSSStyleValue> Parser::parse_single_shadow_value(TokenStream<ComponentVal
     return ShadowStyleValue::create(color.release_nonnull(), offset_x.release_nonnull(), offset_y.release_nonnull(), blur_radius.release_nonnull(), spread_distance.release_nonnull(), placement.release_value());
 }
 
+RefPtr<CSSStyleValue> Parser::parse_rotate_value(TokenStream<ComponentValue>& tokens)
+{
+    // Value:	none | <angle> | [ x | y | z | <number>{3} ] && <angle>
+
+    // "none"
+    if (tokens.remaining_token_count() == 1) {
+        auto transaction = tokens.begin_transaction();
+        if (auto keyword = parse_keyword_value(tokens)) {
+            if (keyword->to_keyword() == Keyword::None) {
+                transaction.commit();
+                return keyword;
+            }
+        }
+    }
+
+    // <angle>
+    if (auto angle = parse_angle_value(tokens)) {
+        // FIXME: Handle calc()
+        if (angle->is_angle())
+            return RotationStyleValue::create(angle->as_angle().angle(), 0, 0, 1);
+    }
```

Comment:
The `[ x | y | z | <number>{3} ] && <angle>` form can have the `<angle>` first, (any `||` or `&&` in the grammar means it happens in any order,) which this would parse incorrectly.